### DISCUSSION TITLE
Cant connect HA to the inverter

### DISCUSSION THREAD

**Discussion by max10019679** [wlcrs/huawei_solar#599] [Connecting to the inverter]
Hi hoping someone can help. I have followed the network connection guide as per https://github.com/wlcrs/huawei_solar/wiki/Connecting-to-the-inverter#network-connection and get stuck at ‘Getting connectivity between HA on your home network and the inverter AP’. I’m using the described straightforward approach – built in wi-fi on the Rasberry Pi.

I go to settings>system>network>wlan0>wifi>scan for access points. The built in Rasberry Pi wifi can see the sun 2000 network available to connect with a signal strength between 50 and 60. I select wpa-psk security and enter the correct password. Every attempt I make I get the message ‘Failed to change network settings Activating connection failed, check connection settings.’

Any help or advice much appreciated. Hope that I have provided enough information below on the set up.

Sun2000-6ktl-L1
Gateway address 192.168.200.1
Inverter firmware V200R001C00SPC130
Battery firmware V100R002C00SPC119
MODBUS firmware V100R001C00B021
Huawei Dongle connected to home network with cable with DHCP and Local O&M enabled in the sun2000 app as installer. Modbus TCP connection is set to ‘enabled (unrestricted)’
Rasberry Pi 4 connected to home network with cable
Home Assistant core 2024.1.5, supervisor 2023.12.1, OS 11.4, Frontend 20240104.0

**Comment by Roving-Ronin** [comment]
@max10019679 

MODBUS firmware V100R001C00B021 is a rather old version, might be worthwhile logging into FusionSolar Portal and updating it.
Login with the Installer level account and go to Plants ---> Update Management --> Create Update Task, and select MBUS and your site, like:

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/e4b77e53-40d6-4a3b-b981-e8c9322f2d71)

Choose the latest MBUS version available for your inverter. You might also want to update the dongle, inverter and battery firmware whilst your at it. For future reference, not that updating the inverter etc does not also update the modbus version, that (as shown here) is a seperate job to run.

Just re-reading your post and it also sounds like you've for the sDongle connected to the home network and the RPi4 also on the LAN (both via RJ-45), so why are you trying to connect to the inverters 'Built in'  WiFi Access Point ?

Give the sDongle a fixed IP address (just assigned it a 'reserved' IP on your home DHCP server) and then you just install the WLCRS integration and when asked for the network address of the inverter, enter the static IP you assigned to the sDongle !

**Reply by max10019679** [reply]
thanks for your reply. whenever I go to update the modbus it keeps saying 'the target version is earlier than the current version. The upgrade may fail.' it says the same for the inverter and battery updates (but why is it offering me updates that are old). Should I definitely proceed and ignore this message? i've attached a screenshot from the modbus update. is there a particular order i should do the updates?

correct the huawei dongle and the RPi have a RJ45 cabled connection to the home network router. from my understanding of the connection guide I need a seperate network to create the bridge? indeed the guide says 'The most straightforward approach is to use an USB WiFi stick, or use the built-in WiFi or your Home Assistant server (for instance: the WiFi functionality of your Raspberry Pi 4). This must be a dedicated WiFi device:'

if i understand you - I should make the updates as you say, give a static IP to the dongle on my home router settings, and then skip to the WLCRS HACS integration instructions?

thanks for your help and advice
![fusion solar update](https://github.com/wlcrs/huawei_solar/assets/157296649/069270c1-8740-4bbd-93f1-64445733fdc9)

**Comment by jires** [comment]
@max10019679 
> I select wpa-psk security and enter the correct password.

How you ensure that password is correct? Can you connect to inverter's wifi with f.x your mobile phone using this password?
BTW. factory password is " Changeme "

@Roving-Ronin 

> Just re-reading your post and it also sounds like you've for the sDongle connected to the home network and the RPi4 also on the LAN (both via RJ-45), so why are you trying to connect to the inverters 'Built in' WiFi Access Point ?

I can see the point in this setup, right now I'm using hardwired sdongle in the home network (for FusionSolar and app), AND inverter's wifi for HA integration. Sdongle restarts or something similar once at 24 hours, and I had a periods of 15-40 min, every 24 hours where integration missed connection almost all the time. When I'm using direct wifi to inverter, no connections issues, it is just rock-solid.

**Reply by max10019679** [reply]
thanks for your reply - yes im certain the password is correct. I guess i was wondering if its to do with the security type - wpa-psk on HA but I think the inverter is wpa-psk 2?. are you using RPi with wifi to the inverter and managed to scan for and connect to the inverter access point?

**Reply by Roving-Ronin** [reply]
@max10019679 
Sorry in bed sick atm. There is no reason to have sDongle for the FusionSolar and seperately using the inverters inbuilt WiFi AP for HA/WLCRS.

The sDongle gives the inverter (or a cascade, plus the DTSU-666H and any batteries) network connectivity. Via this single connection is it able to communicate with the FusionSolar portal and also provide modbus-tcp access (what HA/WLCRS connects to/via). 
Note: This is what my cascade of inverters is doing, 1 sDongle that publishes to FusionSolar and HA / WLCRS also connects to it.

Re the restarts, that appears to be something specific to your situation, I'm checking my HA history and I have no restarts of the sDongle and/or data lossses.

Re the firmware check the list of firmware that others have used and found working and patch to that level. I'd patch the inverter first, then MBUS and sDongle.  https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware

Here's one that seems to fit your scenario:

SUN2000-5KTL-L1 : V200R001C00SPC133 
SDongleA-05 : V100R001C00SPC130 
Connected via Option B (using the sDongle on port 502). Primary Inverter ID# 1
Dongle Ethernet via home network

If you dont have access to the Huawei Solar Support site https://support.huawei.com/enterprise/en/fusionsolar/sun2000l-pid-22027611 then have a look at this EU reseller who mirrors the latest versions https://photomate.zendesk.com/hc/en-gb/categories/4406398674589-Firmware. As shown there V100R001C00SPC330 is the latest modbus firmware, that was released about a year ago.  There's also copies of the SPC136 inverter firmware and the sDongle firmware that you can read the documentation from. See what FusionSolar portal 'shows' as the latest available firmware and be aware that the published firmware there isn't always the latest, for the Australian servers for instance I've had to ring a Huawei contact and get them to log a request to have someone in Global update the firmware available upon the servers (if/when there's a new version that has desired new features / bug fixes).

**Reply by max10019679** [reply]
I managed to update my modbus to SPC300 and I'm now focusing on integrating with the RJ45 connections between the dongle and router and Rpi and router. I logged into my router and made the IP for the dongle static. I then went onto terminal to test the connection with ping. the ping appears successful but the dongle constantly pings back? see attached screenshot:
![ping](https://github.com/wlcrs/huawei_solar/assets/157296649/3e701968-8766-4419-b45d-4ccc83f16277)
I tried to complete the integration by network using the static IP as host, port 502 and slave ID 1 with elevated permissions but no success. same with port 6607:
![network configure](https://github.com/wlcrs/huawei_solar/assets/157296649/93bc0022-1705-448b-9883-6c8031a5dc0b)

appreciate any help or advice. get well soon.

**Reply by Roving-Ronin** [reply]
You have logged into the inverter using FusionSolar app on your mobile phone (with installer login) and confirmed that post upgrades that Modbus tcp is (re)enabled upon the dongle? And checked the ID# it's using?

**Reply by max10019679** [reply]
i just checked the confirmed firmware page https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware#confirmed-firmware-versions and it doesnt look like the Sun2000-6ktl-L1 with a smart dongle is supported?

**Reply by Roving-Ronin** [reply]
As I mentioned above there is one about half way down.

A 5kW L1 with sdongle. Search for '1,16' (Control F) and you should find it.

**Comment by max10019679** [comment]
Yes - it is enabled and unrestricted with ID 1:

[image0.png]
[image1.png]Sent from my iPhone

On 24 Jan 2024, at 12:34, Roving Ronin ***@***.***> wrote:

﻿

You have logged into the inverter using FusionSolar app on your mobile phone (with installer login) and confirmed that post upgrades that Modbus tcp is (re)enabled upon the dongle? And checked the ID# it's using?
